tests/test_BlkDiagMatrix.py

```diff
@@ -0,0 +1,910 @@
+                      [-41.82122077, -42.43246999]]),
+            np.array([[-41.99703348, -41.67601533],
+                      [-41.67601533, -42.12518209]]),
+            np.array([[-41.99703348, -41.67601533],
+                      [-41.67601533, -42.12518209]]),
+            np.array([[-42.06749134, -41.62547766],
+                      [-41.62547766, -41.8210907 ]]),
+            np.array([[-42.06749134, -41.62547766],
+                      [-41.62547766, -41.8210907 ]]),
+            np.array([[-42.13551364]]),
+            np.array([[-42.13551364]]),
+            np.array([[-42.18975111]]),
+            np.array([[-42.18975111]]),
+            np.array([[-42.22661312]]),
+            np.array([[-42.22661312]])]
+
+        # a-b = -1*(b-a)
+        result_2 = [-1*x for x in result_1]
+
+        blk_c = self.blk_a - 42.
+        self.allallfunc(blk_c, result_1)
+
+        # and test the right add
+        blk_c = 42. - self.blk_a
+        self.allallfunc(blk_c, result_2)
+
+    def testBlkDiagMatrixDeepCopy(self):
+        blk_a_copy_1 = self.blk_a.copy()
+        blk_a_copy_2 = self.blk_a.copy()
+
+        # test same values as blk_a
+        self.allallfunc(blk_a_copy_1, self.blk_a)
+        self.allallfunc(blk_a_copy_2, self.blk_a)
+
+        # change a copy, test that copy is changed
+        blk_a_copy_1 *= 2.
+        self.allallfunc(blk_a_copy_1, self.blk_a,
+                        func=lambda x,y: not np.allclose(x,y))
+        self.allallfunc(blk_a_copy_1, blk_a_copy_2,
+                        func=lambda x,y: not np.allclose(x,y))
+        # and blk_a is unchanged
+        self.allallfunc(blk_a_copy_2, self.blk_a)
+
+    def testBlkDiagMatrixInPlace(self):
+        """ Tests sequence of in place optimized arithmetic (add, sub, mul) """
+        result_1 = [x + x + 10. for x in self.blk_a]
+        result_2 = [np.ones(x.shape) * 5. for x in self.blk_a]
+
+        # make a block diagonal object to mutate
+        blk_c = self.blk_a.copy()
+        # store the python object ids of each array in blk_c
+        #  we want to ensure the actual object refs are _not_ changing
+        #  for in place operations.
+        id0 = [id(x) for x in blk_c]
+
+        blk_c += self.blk_a
+        self.assertTrue(all(id(blk_c[x]) == id0[x] for x in range(len(blk_c))))
+
+        blk_c += 10.
+        self.assertTrue(all(id(blk_c[x]) == id0[x] for x in range(len(blk_c))))
+
+        blk_a5 = BlkDiagMatrix.ones(self.blk_partition)
+        id1 = [id(x) for x in blk_a5]
+        blk_a5 *= 5.
+        self.assertTrue(all(id(blk_a5[x]) == id1[x] for x in range(len(blk_c))))
+
+        blk_c -= blk_a5
+        blk_c -= blk_a5
+        self.assertTrue(all(id(blk_c[x]) == id0[x] for x in range(len(blk_c))))
+
+        blk_c -= self.blk_a
+        self.assertTrue(all(id(blk_c[x]) == id0[x] for x in range(len(blk_c))))
+
+        # a = (a+a+10) - 5. - 5. -a
+        self.allallfunc(blk_c, self.blk_a)
+
```

Comment:
Remove.

Reply:
I removed the comment. Let me know if you didn't like the arithmetic test and I'll change it. Thanks

Reply:
Ahh, it was the extra line, I see. got it.